mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -111,8 +111,15 @@ public void configureTableJobProperties(TableDesc tableDesc, Map<String, String>
   // @Override
   public void configureInputJobCredentials(TableDesc tableDesc, Map<String, String> secrets) {}
 
+  private void setCommonJobConf(JobConf jobConf) {
```

Comment:
What if the configuration is already set with different values?

Reply:
according to https://issues.apache.org/jira/browse/HIVE-25581, this change should ensure it's set by the Iceberg storage handler by default.  
if its value is set with different values,  the Iceberg storage handler seems not able to work normally.

Reply:
My question was aimed to:
- What do we do if the `tez.mrreader.config.update.properties` already contains some values?

The configuration seems like a comma separated list. Simply removing the old values seems like a bad idea. Maybe adding them at the end of the list, or adding them, if they are not yet there could be a better approach.

Reply:
hi @pvary , thanks for your comments.
 I have adapted the code to make sure that the setting includes both the old values and the `hive.io.file.readcolumn.names,hive.io.file.readcolumn.ids`.  
do you have any further sugguestions? 
thanks

Reply:
Thanks @ludlows!
As I mentioned in another comment, could you please add some unit tests to ensure that this feature keeps working?

Reply:
Hi @pvary , thanks for your suggestions. 
now I have implemented one unit test.
however, according to the test result, it seems that my modification doesn't take effects. 
could you give me a hint about the test  failure ?
many thanks

Reply:
I think the changes effect of the configuration used by the query, and do not effect the configuration used by the shell.
You could check the values on the executors when the job is running. Or you could check the result of the query. Create a test which fails without the change because of the wrong pruning, and successful after the change

Reply:
@pvary it seems that I need to create a table without setting `tez.mrreader.config.update.properties` and execute a query sql getting a wrong result. but where could I find such table ?  or my understanding is wrong ?

Reply:
@ludlows: The tez config is not table level. It is an engine level configuration, so you have to set it in your test somewhere around the HiveShell start/configuration